src/ripe/base.py

```diff
@@ -54,6 +54,83 @@ class API(
     transport_rule.TransportRuleAPI,
     availability_rule.AvailabilityRuleAPI,
 ):
+    @classmethod
+    def _query_to_spec(cls, query):
+        options = cls._unpack_query(query)
+        brand = options.get("brand", None)
+        model = options.get("model", None)
+        variant = options.get("variant", None)
+        version = options.get("version", None)
+        description = options.get("description", None)
+        initials = options.get("initials", None)
+        engraving = options.get("engraving", None)
+        initials_extra = options.get("initials_extra", [])
+        tuples = options.get("p", [])
+        tuples = tuples if isinstance(tuples, list) else [tuples]
+        initials_extra = (
+            initials_extra if isinstance(initials_extra, list) else [initials_extra]
+        )
+        initials_extra = cls._parse_extra_s(initials_extra)
+        parts = cls._tuples_to_parts(tuples)
+        parts_m = cls._parts_to_parts_m(parts)
+        spec = dict(
+            brand=brand,
+            model=model,
+            parts=parts_m,
+            initials=initials,
+            engraving=engraving,
+            initials_extra=initials_extra,
+        )
+        if variant:
+            spec["variant"] = variant
+        if version:
+            spec["version"] = version
+        if description:
+            spec["description"] = description
+        return spec
+
+    @classmethod
+    def _unpack_query(cls, query):
+        query = query[1:] if query[0] == "?" else query

```

Comment:
assuming we're receiving a valid query, I think we can just strip for the same effect

```suggestion
        query = query.strip("?")

```